[FSDP2] Fixed 2D clip grad norm test

awgu opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #126497

This fixes https://github.com/pytorch/pytorch/issues/126484.

We change from transformer to MLP stack since transformer seems to introduce slight numeric differences when using TP. We include a sequence parallel layer norm module in the MLP stack to exercise `(S(0), R)` placement.

cc @mrshenli @pritamdamania87 @zhaojuanmao @satgera @gqchen @aazzolini @osalpekar @jiayisuse @H-Huang @kwen2501 @penguinwu @fegin @XilunWu @wanchaol @fduwjj @wz337 @tianyu-l @wconstab @yf225 @chauhang @d4l3k

Commit:
[FSDP2] Fixed 2D clip grad norm test

[ghstack-poisoned]

Commit:
Update on "[FSDP2] Fixed 2D clip grad norm test"




cc mrshenli pritamdamania87 zhaojuanmao satgera gqchen aazzolini osalpekar jiayisuse H-Huang kwen2501 penguinwu fegin XilunWu wanchaol fduwjj wz337 tianyu-l wconstab yf225 chauhang d4l3k

[ghstack-poisoned]

Commit:
Update on "[FSDP2] Fixed 2D clip grad norm test"


This fixes https://github.com/pytorch/pytorch/issues/126484.

We change from transformer to MLP stack since transformer seems to introduce slight numeric differences when using TP. We include a sequence parallel layer norm module in the MLP stack to exercise `(S(0), R)` placement.

cc mrshenli pritamdamania87 zhaojuanmao satgera gqchen aazzolini osalpekar jiayisuse H-Huang kwen2501 penguinwu fegin XilunWu wanchaol fduwjj wz337 tianyu-l wconstab yf225 chauhang d4l3k

[ghstack-poisoned]

weifengpy commented:
switching to fsdp2 release note is really some labor work

wz337 reviewed (approved):
LGTM! Thanks for looking into it so quickly.

awgu commented:
Failure are all inductor-related, not FSDP2-related.

awgu commented:
@pytorchbot merge -i

jeanschmidt commented:
@pytorchbot revert -m "reverting to check if might have introduced inductor cuda 12 issues" -c "ignoredsignal"

awgu commented:
@jeanschmidt I wonder if we can re-land this. I do not think it is related to Inductor CUDA 12 failures.

awgu commented:
@pytorchbot rebase -s

Commit:
Update

[ghstack-poisoned]

Commit:
Update on "[FSDP2] Fixed 2D clip grad norm test"


This fixes https://github.com/pytorch/pytorch/issues/126484.

We change from transformer to MLP stack since transformer seems to introduce slight numeric differences when using TP. We include a sequence parallel layer norm module in the MLP stack to exercise `(S(0), R)` placement.

cc mrshenli pritamdamania87 zhaojuanmao satgera gqchen aazzolini osalpekar jiayisuse H-Huang kwen2501 penguinwu fegin XilunWu wanchaol fduwjj wz337 tianyu-l wconstab yf225 chauhang d4l3k

[ghstack-poisoned]

awgu commented on .ci/pytorch/test.sh:
```diff
@@ -326,6 +326,7 @@ test_inductor_distributed() {
   python test/run_test.py -i distributed/_composable/fsdp/test_fully_shard_frozen.py --verbose
   python test/run_test.py -i distributed/_composable/fsdp/test_fully_shard_mixed_precision.py -k test_compute_dtype --verbose
   python test/run_test.py -i distributed/_composable/fsdp/test_fully_shard_mixed_precision.py -k test_reduce_dtype --verbose
+  python test/run_test.py -i distributed/_composable/fsdp/test_fully_shard_clip_grad_norm_.py -k test_clip_grad_norm_2d --verbose
```
Test name was incorrect before (fixed to include trailing `_`).

awgu commented:
@pytorchbot merge

awgu commented:
@pytorchbot rebase -s

awgu commented:
@pytorchbot merge
